agent/battle/runtime/runtime.py

```
@@ -225,9 +226,41 @@ def _observe(self):
+        # 有 Chaldea 计划时：技能 CD 检测无意义（计划已精确指定何时放技能），
+        # 且 ColorMatch ROI 可能因分辨率/缩放偏移导致误判为 CD。
+        # 将所有技能标记为可用，避免被安全门过滤掉。
+        has_plan = hasattr(self.decider, 'plan') and self.decider.plan is not None
+        if has_plan:
+            result = self._patch_skills_available(result)
+            mfaalog.info("[AutoBattle] _observe() -> plan active, all skills patched to available")
+
         mfaalog.info(f"[AutoBattle] _observe() -> perception built, scene={result.scene.name}")
         return result
 
+    def _patch_skills_available(self, state: BattleState) -> BattleState:
```

Ruff (0.16.1)

[error] Undefined name `BattleState`

(F821)